### PR TITLE
RSDK-6983 Allow users to disable metrics through config

### PR DESCRIPTION
**Pull request by ale7714**
[RSDK-6983](https://viam.atlassian.net/browse/RSDK-6983)

Decided to go directly to use attributes from config bc it's way easier for the users. Tested on macos and raspi 4. 

[RSDK-6983]: https://viam.atlassian.net/browse/RSDK-6983?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ale7714** [comment]
Looks like a big PR because I broke down the config into sections but most of the code change is within telegafconfig.go . Getting ready for eventually allowing customer to customize specific metrics.

**Review by cheukt — CHANGES_REQUESTED** [review]
did a quick pass - some thoughts

- can you update the readme for the new config structure?
- does reconfiguration work?
- would it make sense to disable a few fields by default (e.g. wifi), and tell people how to enable it in the readme?

**Comment by ale7714** [comment]
> * can you update the readme for the new config structure?

Done! 

> * does reconfiguration work?

With viam-server running, I update the cloud configuration and see the module reconfigure and additionally validating that the contents of `/tmp/viam-telegraf.conf` has properly updated

```
2024-03-26T18:46:52.011-0400	INFO	robot_server.process.telegraf-sensor_/Users/aleparedes/viam/viam-telegraf-sensor/run.sh.StdOut	pexec/managed_process.go:244	
\_ 2024-03-26T22:46:52.010Z	DEBUG	telegraf-sensor	module/module.go:410	rebuilding	{"name":"rdk:component:sensor/sensorT"}
```

> * would it make sense to disable a few fields by default (e.g. wifi), and tell people how to enable it in the readme?

The two metrics I can think could cause issues are `wireless` and `temp`. Any other that you think is worth disabling by default?

File: telegrafsensor/telegrafconfig.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	err = os.WriteFile(telegrafConfPath, baseConfigData, 0644)
+	if err != nil {
+		return fmt.Errorf("error writing config file: %v", err)
+	}
+
+	emptyInputs := true
+
+	for confName, disableField := range metricMap {
+		switch disableField {
+		case "disable_temp", "disable_wireless":
```
Comment (cheukt) [inline telegrafsensor/telegrafconfig.go:39]: add a comment about why temp/wireless is disabled by default?

**Review by cheukt — APPROVED** [review]
LGTM

**Pull request merged**